vignettes/datatable-reference-semantics.Rmd

```diff
@@ -112,6 +112,8 @@ Note that the code above explains how `:=` can be used. They are not working exa
 
 * On the other hand, (b) is handy if you would like to jot some comments down for later.
 
+* In general, (a) and (b) should produce the same result, however there are minor differences in certain cases due to the implementation details of `:=`. Specifically, when using (a), simple assignments like `LHS := RHS` wrap `RHS` into a list automatically, making it easier for users to assign a single column.
```

Comment:
there are minor differences... Specifically, ... makes me think that you are going to explain a specific example of one of the minor differences, but I do not see any example/explanation, can that be added? (would be useful for my understanding at least)

Reply:
Yes, I changed it around a little, check out the most updated commit